controllers/storagecluster/reconcile.go

```diff
@@ -205,20 +205,26 @@ func (r *StorageClusterReconciler) validateStorageClusterSpec(instance *ocsv1.St
 	if err := versionCheck(instance, r.Log); err != nil {
 		r.Log.Error(err, "Failed to validate version")
 		r.recorder.Event(instance, statusutil.EventTypeWarning, statusutil.EventReasonValidationFailed, err.Error())
+		instance.Status.Phase = statusutil.PhaseError
+		_ = r.Client.Update(context.TODO(), instance)
```

Comment:
Why are you ignoring this error? I generally don't like that.

Reply:
Because anyway we are going to return another error either it fails or succeeds. I am doing my best effort here. Moreover IMO the Spec error is much more important than the update error.

Reply:
It is important to know all errors that occurred, otherwise we may inhibit proper troubleshooting of unforeseen failure cases. If something like `Update()` fails we probably have much bigger problems than an invalid version. You already logged the `versionCheck()` err and recorded the Event, so you can either return the `Update()` err if it is not `nil` otherwise return the `versionCheck()` err.

Reply:
Done